### PR TITLE
switch to new jit_builder.

### PR DESCRIPTION
**Pull request by xuhancn**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118515
* #118514

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

Differential Revision: [D53203201](https://our.internmc.facebook.com/intern/diff/D53203201)

### CODE REVIEW

File: torch/_inductor/codecache.py [hunk 1]
```diff
@@ -1079,7 +1009,62 @@ def __bool__(self) -> bool:  # type: ignore[override]
...
+
+    cpp_code = get_x86_isa_detect_code()
+
+    key, input_path = write(cpp_code, "cpp")
+
+    from filelock import FileLock
+
+    lock_dir = get_lock_dir()
+    lock = FileLock(os.path.join(lock_dir, key + ".lock"), timeout=LOCK_TIMEOUT)
+    with lock:
```
Comment (Skylion007) [inline torch/_inductor/codecache.py:1039]:
```suggestion
    with FileLock(os.path.join(lock_dir, key + ".lock"), timeout=LOCK_TIMEOUT) as lock:
```

File: torch/_inductor/codecache.py [hunk 2]
```diff
@@ -1834,31 +1882,37 @@ def _load_library(cls, path: str, key: str) -> Union[CDLL, ModuleType]:
...
             )
         )
         key, input_path = write(source_code, "cpp", extra=cpp_command)
+        """
         if key not in cls.cache:
             from filelock import FileLock
 
             lock_dir = get_lock_dir()
             lock = FileLock(os.path.join(lock_dir, key + ".lock"), timeout=LOCK_TIMEOUT)
             with lock:
```
Comment (Skylion007) [inline torch/_inductor/codecache.py:1907]: same here

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53203201).

File: torch/_inductor/codecache.py (hunk 2 above)
Comment (xuhancn) [reply torch/_inductor/codecache.py]: Done

**Pull request closed without merging**